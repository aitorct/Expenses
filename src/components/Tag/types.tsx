import {StyleProp, TextStyle, ViewStyle} from 'react-native';

interface BaseTag {
  textStyle?: StyleProp<TextStyle>;
  style?: StyleProp<ViewStyle>;
}

interface IconTextTag extends BaseTag {
  icon: string;
  text: string | number;
}

interface IconTag extends BaseTag {
  icon: string;
  text?: never;
}

interface TextTag extends BaseTag {
  text: string | number;
  icon?: never;
}

export type TagType = IconTextTag | IconTag | TextTag;
