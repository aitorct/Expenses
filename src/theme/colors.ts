const colors = {
  black: '#000000',
  offwhite: '#F3F4F6',
  lightGrey: '#7D8285',
  white: '#FFFFFF',
  lightGrayishBlue: '#E7EFFA',
  darkGrayishBlue: '#74a2e2',
};

export default {
  background: colors.offwhite,
  secondaryBackground: colors.white,
  tertiaryBackground: colors.black,
  icon: colors.black,
  text: colors.black,
  secondaryText: colors.white,
  placeholderText: colors.lightGrey,
  categoryText: colors.lightGrey,
  shadow: colors.black,
  borderColor: colors.black,
  selectionColor: colors.black,
  textInputBackground: `${colors.lightGrey}30`,
  tagBackground: colors.lightGrayishBlue,
  touchableTagBackground: `${colors.lightGrey}1A`,
  selectedTouchableTagBorder: colors.darkGrayishBlue,
};
