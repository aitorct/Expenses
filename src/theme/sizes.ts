const sizes = {
  s1: 1,
  s4: 4,
  s8: 8,
  s10: 10,
  s12: 12,
  s16: 16,
  s24: 24,
  s32: 32,
  s36: 36,
  s48: 48,
  s100: 100,
};

export default {
  ...sizes,
  smallIcon: sizes.s16,
  icon: sizes.s24,
  border: sizes.s1,
};
