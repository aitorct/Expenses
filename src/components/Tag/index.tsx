import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import colors from '@theme/colors';
import sizes from '@theme/sizes';
import {TagType} from './types';

const Tag = ({icon, text, textStyle, style}: TagType) => (
  <View style={[styles.tag, style]}>
    {icon && <Icon name={icon} color={colors.text} size={sizes.smallIcon} />}
    {icon && text && <View style={styles.separator} />}
    {text && (
      <View style={styles.textWrapper}>
        <Text style={[styles.text, textStyle]}>{text}</Text>
      </View>
    )}
  </View>
);

const styles = StyleSheet.create({
  tag: {
    flexDirection: 'row',
    backgroundColor: colors.tagBackground,
    padding: sizes.s4,
    paddingHorizontal: sizes.s8,
    borderRadius: sizes.s36,
    alignItems: 'center',
  },
  separator: {
    paddingLeft: sizes.s4,
  },
  textWrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  text: {
    color: colors.text,
  },
});

export default Tag;
