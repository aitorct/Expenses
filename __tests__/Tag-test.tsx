import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Tag from '@components/Tag';
import sizes from '@theme/sizes';
import colors from '@theme/colors';

it('Tag with text renders correctly', () => {
  const tree = renderer.create(<Tag text={'Tag test'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Tag with icon renders correctly', () => {
  const tree = renderer.create(<Tag icon={'eye'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Tag with icon and text renders correctly', () => {
  const tree = renderer.create(<Tag icon="eye" text={'Tag test'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Tag with style renders correctly', () => {
  const style = {padding: sizes.s24};

  const tree = renderer
    .create(<Tag text={'Tag test'} style={style} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Tag with text style renders correctly', () => {
  const textStyle = {color: colors.placeholderText};

  const tree = renderer
    .create(<Tag text={'Tag test'} textStyle={textStyle} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
