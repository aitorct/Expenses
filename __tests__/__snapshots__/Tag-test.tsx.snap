// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag with icon and text renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E7EFFA",
        "borderRadius": 36,
        "flexDirection": "row",
        "padding": 4,
        "paddingHorizontal": 8,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 16,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰈈
  </Text>
  <View
    style={
      Object {
        "paddingLeft": 4,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          undefined,
        ]
      }
    >
      Tag test
    </Text>
  </View>
</View>
`;

exports[`Tag with icon renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E7EFFA",
        "borderRadius": 36,
        "flexDirection": "row",
        "padding": 4,
        "paddingHorizontal": 8,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 16,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰈈
  </Text>
</View>
`;

exports[`Tag with style renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E7EFFA",
        "borderRadius": 36,
        "flexDirection": "row",
        "padding": 4,
        "paddingHorizontal": 8,
      },
      Object {
        "padding": 24,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          undefined,
        ]
      }
    >
      Tag test
    </Text>
  </View>
</View>
`;

exports[`Tag with text renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E7EFFA",
        "borderRadius": 36,
        "flexDirection": "row",
        "padding": 4,
        "paddingHorizontal": 8,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          undefined,
        ]
      }
    >
      Tag test
    </Text>
  </View>
</View>
`;

exports[`Tag with text style renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E7EFFA",
        "borderRadius": 36,
        "flexDirection": "row",
        "padding": 4,
        "paddingHorizontal": 8,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          Object {
            "color": "#7D8285",
          },
        ]
      }
    >
      Tag test
    </Text>
  </View>
</View>
`;
